R/module_transform_data.R

```diff
@@ -40,40 +48,94 @@ ui_transform_data <- function(id, transformers = list(), class = "well") {
         tags$i(
           class = "remove pull-right fa fa-angle-down",
           style = "cursor: pointer;",
-          title = "fold/expand transform panel",
-          onclick = sprintf("togglePanelItems(this, '%s', 'fa-angle-right', 'fa-angle-down');", wrapper_id)
+          title = "fold/expand transformator panel",
+          onclick = sprintf("togglePanelItems(this, '%s', 'fa-angle-right', 'fa-angle-down');", transform_wrapper_id)
         ),
-        div(
-          id = wrapper_id,
-          ui_teal_data(id = ns(name), data_module = transformers[[name]]$ui)
+        tags$div(
+          id = transform_wrapper_id,
+          if (is.null(data_mod$ui)) {
+            return(NULL)
```

Comment:
why `return()`?
```suggestion
           NULL
```

Reply:
This is a badly placed early return clause to completely hide UI elements, but this raises an issue.

In the case of an error on transform when there is no UI (not even `function(id) NULL`) it should still show error.

Implemented in 0f9565c62

Reply:
### New & old when it shouldn't be shown

![image](https://github.com/user-attachments/assets/f2cb1466-db82-4e49-b482-014f75addc59)

### New

![image](https://github.com/user-attachments/assets/dfb9aef4-39e9-46ad-a91a-798be8b4fa20)

### Old

![image](https://github.com/user-attachments/assets/5ad86045-0381-4250-84b2-6bd6771a1501)

Reply:
It also begs the question, should we completely hide it